### PR TITLE
use timeout for lock time control

### PR DESCRIPTION
**Pull request by gabrielbazan7**

### CODE REVIEW

File: src/pages/pin/pin.ts [hunk 1]
```diff
@@ -48,15 +51,10 @@ export class PinModalPage {
...
-        let config = this.configProvider.get();
-        let bannedUntil = config.lock.bannedUntil;
-        if (bannedUntil) {
-          let now = Math.floor(Date.now() / 1000);
-          if (now < bannedUntil) {
-            this.disableButtons = true;
-            this.lockTimeControl(bannedUntil);
-          }
-        }
+        this.persistenceProvider.getLockApp().then((val: string) => {
```
Comment (matiu) [inline src/pages/pin/pin.ts]: don't use `val` as a variable name. maybe `isLocked`

File: src/providers/persistence/persistence.ts [hunk 2]
```diff
@@ -511,4 +511,15 @@ export class PersistenceProvider {
     return this.storage.get('order-' + walletId);
   };
 
+  setLockApp(locked: string): Promise<void> {
+    return this.storage.set('lockApp', locked);
```
Comment (matiu) [inline src/providers/persistence/persistence.ts]: `lockApp` => maybe: `isLocked` ?

Comment (matiu) [reply src/providers/persistence/persistence.ts]: then, +    return this.storage.set('isLocked', isLocked);

File: src/pages/pin/pin.ts (hunk 1 above)
Comment (matiu) [inline src/pages/pin/pin.ts]: getLockApp => . getLockStatus

File: src/pages/pin/pin.ts [hunk 3]
```diff
@@ -100,35 +98,31 @@ export class PinModalPage {
...
       this.expires = ('0' + m).slice(-2) + ":" + ('0' + s).slice(-2);
-    }
+    }, 1000);
+
+    setTimeout(() => {
+      this.reset(countDown);
+    }, this.ATTEMPT_LOCK_OUT_TIME * 1000);
   }
 
   private reset(countDown) {
```
Comment (matiu) [inline src/pages/pin/pin.ts]: reset => unlock

File: src/pages/pin/pin.ts [hunk 4]
```diff
@@ -100,35 +98,31 @@ export class PinModalPage {
...
-    }, 1000);
-  }
-
-  private setExpirationTime(bannedUntil, countDown) {
-    let now = Math.floor(Date.now() / 1000);
-    if (now > bannedUntil) {
-      if (countDown) this.reset(countDown);
-    } else {
-      this.disableButtons = true;
+  private lockTimeControl() {
```
Comment (matiu) [inline src/pages/pin/pin.ts]: break in 2 functions:
1.- showLocktimer
2.- setLockRelease

File: src/pages/pin/pin.ts (hunk 3 above)
Comment (matiu) [reply src/pages/pin/pin.ts]: reset should not receive any .params.

File: src/pages/pin/pin.ts [hunk 5]
```diff
@@ -100,35 +98,31 @@ export class PinModalPage {
...
+      let now = Math.floor(Date.now() / 1000);
       let totalSecs = bannedUntil - now;
       let m = Math.floor(totalSecs / 60);
       let s = totalSecs % 60;
       this.expires = ('0' + m).slice(-2) + ":" + ('0' + s).slice(-2);
-    }
+    }, 1000);
+
+    setTimeout(() => {
+      this.reset(countDown);
```
Comment (matiu) [inline src/pages/pin/pin.ts]: clearInterval(countDown);
should be called here.

**Comment by Gamboster** [comment]
LGFM
Tested on my Android

**Comment by matiu** [comment]
ACK!

**Pull request merged**